File: app/src/main/java/com/example/final_pocproject/api/ApiInterface.java
package com.example.final_pocproject.api;

import com.example.final_pocproject.model.Comments;
import com.example.final_pocproject.model.Post;
import com.example.final_pocproject.model.User;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiInterface {

    @GET("users")
    Call<List<User>> getUsers();

    @GET("users/{id}/posts")
    Call<List<Post>> getPosts(@Path("id") int postId);

    @GET("posts/{id}/comments")
    Call<List<Comments>> getComments(@Path("id") int postId);

}

File: app/src/main/java/com/example/pocProjectKotlin/retrofit/ApiInterface.kt
package com.example.pocProjectKotlin.retrofit

import com.example.pocProjectKotlin.model.Comment
import com.example.pocProjectKotlin.model.Post
import com.example.pocProjectKotlin.model.User
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiInterface {

    @GET("users")
    fun getUsers(): Call<List<User>>

    @GET("users/{id}/posts")
    fun getPosts(@Path("id") userId: Int): Call<List<Post>>

    @GET("posts/{id}/comments")
    fun getComments(@Path("id") postId: Int): Call<List<Comment>>
}